### DISCUSSION TITLE
Manipulate and filter output of Get-Account

### DISCUSSION THREAD

**Discussion by sardiskan** [pspete/psPAS#500] [Q&A]
I'm attempting to create a programatic way to login to servers with the root password for each one which is saved and managed in cyberark so I can install an application. To do this I need to be able to associate the accountID with the servername so I can then do a retrieval of the password.

I can get the Get-PASAccount to retrieve the account I need, but it displays all the information I don't need. What is the best way to go about manipulating the data on the fly so I'll only get the address and the password in my output? In standard powershell, a lot of commands create "objects" so you can reference those objects by name. Such as (get-aduser -filter *).name. This would only display the name object of the command's output. The output of the Get-PASAccount doesn't create objects like this so if I do (Get-PASAccount -search username).AccountID I wouldn't get any results because the data output doesn't create objects.

**Comment by pspete** [comment]
Hi @sardiskan 

psPAS makes heavy use of PowerShell objects.
Use the `Get-Member` cmdlet to find available properties of an object which can then be used to filter accordingly
![image](https://github.com/pspete/psPAS/assets/10600090/0074e632-90e7-4ed6-af9c-6db1644ad822)
You can see the property `id` is available on a user object, and can be used to obtain the property value directly.

Some confusion in your scenario probably arises from the output display (`AccountID`) differing from the object property name (`id`). This is likely a hangover from the previous API version - we will look to update in a future version.

**Comment by sardiskan** [comment]
Yes, I just realized before coming back here that the output IS placed into objects, I was just referencing the wrong object. While the output of the basic command says the name of the object is "AccountID"...in the actual data, the object is called just 'id". Thanks for this information and clarification.
